servers/src/grin/dandelion_monitor.rs

```diff
@@ -45,37 +47,110 @@ pub fn monitor_transactions<T>(
 		.spawn(move || {
 			loop {
 				let tx_pool = tx_pool.clone();
-				let stem_transactions = tx_pool.read().unwrap().stem_transactions.clone();
-				let time_stem_transactions = tx_pool.read().unwrap().time_stem_transactions.clone();
 
-				for tx_hash in stem_transactions.keys() {
-					let time_transaction = time_stem_transactions.get(tx_hash).unwrap();
-					let interval = now_utc().to_timespec().sec - time_transaction;
+				let now = now_utc().to_timespec().sec;
 
-					if interval >= config.dandelion_embargo {
-						let source = TxSource {
-							debug_name: "dandelion-monitor".to_string(),
+				let mut fresh_entries = vec![];
+				let mut fluff_stempool = false;
+				{
+					let mut tx_pool = tx_pool.write().unwrap();
+					let mut rng = rand::thread_rng();
+
+					for mut entry in tx_pool.stempool.entries.iter_mut() {
+						//
+						// TODO patience timer config
+						//
+						let cutoff = now - 10;
```

Comment:
Are you planning, to make the patience based on each transaction ?
From what I understood, it should be a repetitive process every ```dandelion_patience``` seconds.
Just to be clear the ```cutoff``` is here because you run the process every 10 seconds instead of every second ? (Also why?)